src/server/src/main/java/io/cassandrareaper/resources/DiagEventSubscriptionResource.java

```diff
@@ -0,0 +1,128 @@
+@Produces(MediaType.APPLICATION_JSON)
+public final class DiagEventSubscriptionResource {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DiagEventSubscriptionResource.class);
+
+  private final AppContext context;
+  private final DiagEventSubscriptionService diagEventService;
+
+  public DiagEventSubscriptionResource(AppContext context, HttpClient httpClient) {
+    this.context = context;
+    this.diagEventService = DiagEventSubscriptionService.create(context, httpClient);
+  }
+
+  @GET
+  public Response getEventSubscriptionList(
+          @QueryParam("clusterName") String clusterName) {
+
+    LOG.debug("get event subscriptions called");
+    Collection<DiagEventSubscription> subscriptions = context.storage.getEventSubscriptions(clusterName);
+    return Response.ok().entity(subscriptions).build();
+  }
+
+  @POST
+  public Response addEventSubscription(
+          @QueryParam("clusterName") String clusterName,
+          @QueryParam("description") Optional<String> description,
+          @QueryParam("includeNodes") String sincludeNodes,
+          @QueryParam("events") String sevents,
+          @QueryParam("exportSse") boolean exportSse,
+          @QueryParam("exportFileLogger") String exportFileLogger,
+          @QueryParam("exportHttpEndpoint") String exportHttpEndpoint) {
+
+    List<String> events = Arrays.asList(sevents == null ? new String[]{} : sevents.split(","));
+    List<String> includeNodes = Arrays.asList(sincludeNodes == null ? new String[]{} : sincludeNodes.split(","));
+
+    DiagEventSubscription subscription = new DiagEventSubscription(
+        Optional.empty(),
+        clusterName,
+        description,
+        includeNodes,
+        events,
+        exportSse,
+        exportFileLogger,
+        exportHttpEndpoint);
+
+    LOG.debug("creating event subscription: {}", subscription);
+    subscription = diagEventService.addEventSubscription(subscription);
+
+    return Response.ok().entity(subscription).build();
+  }
+
+  @GET
+  @Path("/{id}")
+  public Response getEventSubscription(
+          @PathParam("id") UUID id) {
+
+    LOG.debug("get subscription called with id: {}", id);
+    DiagEventSubscription subscription = diagEventService.getEventSubscription(id);
+    if (null == subscription) { //@fixme change to catch
+      return Response.status(Response.Status.NOT_FOUND).build();
+    }
+    return Response.ok().entity(subscription).build();
+  }
+
+  @DELETE
+  @Path("/{id}")
+  public Response deleteEventSubscription(
+          @PathParam("id") UUID id) {
+
+    LOG.debug("deleting subscription with id: {}", id);
+    DiagEventSubscription subscription = diagEventService.getEventSubscription(id);
+    if (null == subscription) { //@fixme change to catch
+      return Response.status(Response.Status.NOT_FOUND).build();
+    }
+
+    assert id.equals(subscription.getId());
+    if (diagEventService.deleteEventSubscription(id)) {
+      return Response.ok().entity(subscription).build();
```

Comment:
rest commands shouldn't return bodies/entities, they should redirect to GET the resource instead. same applies to the post above. i will fix.

(i thought we have a test that enforced this…?)

Reply:
and `observable.js` won't (or can't) follow a redirect to the resource?

Reply:
This has been fixed/improved. This DELETE rest endpoint now returns `202 ACCEPTED`, without an entity body, like other delete commands in Reaper.